modules/system/console/CreateMigration.php

```diff
@@ -192,13 +192,130 @@ protected function prepareVars(): array
+            die($e->getMessage());
+        }
+
+        $modelName = $vars['plugin_namespace'] . '\\Models\\' . $vars['model'];
+
+        $vars['model'] = $model = new $modelName();
+
+        foreach (['fields', 'tabs', 'secondaryTabs'] as $type) {
+            if (!isset($fields[$type])) {
+                continue;
+            }
+            if ($type === 'fields') {
+                $fieldList = $fields[$type];
+            } else {
+                $fieldList = $fields[$type]['fields'];
+            }
+
+            foreach ($fieldList as $field => $config) {
+                if (str_contains($field, '@')) {
+                    list($field, $context) = explode('@', $field);
+                }
+
+                $type = $config['type'] ?? 'text';
+
+                if (str_starts_with($field, '_')
+                    or $field === $model->getKeyName()
+                    or str_contains($field, '[')
+                    or in_array($type, ['fileupload','relation','relationmanager','section','hint'])
+                    or in_array($field, $model->purgeable ?? [])
+                    or $model->getRelationType($field)
+                ) {
+                    continue;
+                }
+
+                $vars['fields'][$field] = $this->mapFieldType($field, $config);
+            }
+        }
+
+        foreach ($model->getRelationDefinitions() as $relationType => $definitions) {
+            if (in_array($relationType, ['belongsTo', 'hasOne'])) {
+                foreach (array_keys($definitions) as $relation) {
+                    $vars['fields'][$relation . '_id'] = [
+                        'type' => 'foreignId',
+                        'index' => true,
+                        'required' => true,
+                    ];
+                }
+            }
+        }
+
+        if ($model->methodExists('getSortOrderColumn')) {
+            $field = $model->getSortOrderColumn();
+            $vars['fields'][$field] = [
+                'type' => 'unsignedinteger',
+                'required' => false,
+                'index' => true,
+            ];
+        }
+
+        $vars['primaryKey'] = $model->getKeyName();
+        $vars['jsonable'] = $model->getJsonable();
+        $vars['timestamps'] = $model->timestamps;
+
+        if ($morphable = $model->morphTo) {
+            $vars['morphable'] = array_keys($morphable);
+        }
+
+        return $vars;
+    }
+
+    protected function mapFieldType($name, $fieldConfig)
+    {
+        switch ($fieldConfig['type'] ?? 'text') {
+            case 'checkbox':
+            case 'switch':
+                $dbType = 'unsignedinteger';
+                break;
+            case 'number':
+            case 'range':
+                $dbType = 'integer';
+                break;
+            case 'datepicker':
+                $dbType = 'timestamp';
+                break;
+            case 'markdown':
+            case 'textarea':
+                $dbType = 'mediumText';
```

Comment:
Should we use `mediumText` here as well or is `text` enough?

Reply:
Let's go `text` for textarea and perhaps `mediumText` for markdown, I reckon. Some markdown documents are mighty large and then you have to include formatting syntax.

Reply:
Done